backend/src/apiserver/main.go

```diff
@@ -72,7 +73,19 @@ func main() {
 		template.Launcher = common.GetStringConfig(launcherEnv)
 	}
 
-	clientManager := cm.NewClientManager()
+	backgroundCtx, backgroundCancel := context.WithCancel(context.Background())
+	defer backgroundCancel()
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
```

Comment:
This add should happen in the `init` function just before launching the goroutine that starts the cache and not here. You could lead to situations where `wg.Done()` is never called otherwise.